pkg/cloud/services/provider/provider.go

```diff
@@ -96,10 +97,19 @@ func NewClient(cloud clientconfig.Cloud, caCert []byte) (*gophercloud.ProviderCl
 	if cloud.Verify != nil {
 		config.InsecureSkipVerify = !*cloud.Verify
 	}
+
 	if caCert != nil {
 		config.RootCAs.AppendCertsFromPEM(caCert)
 	}
 
+	if cloud.CACertFile != "" {
+		certfile, err := os.ReadFile(cloud.CACertFile)
+		if err != nil {
+			return nil, nil, "", fmt.Errorf("can't open cloud.yaml cacert file: %w", err)
+		}
+		config.RootCAs.AppendCertsFromPEM(certfile)
+	}
+
```

Comment:
we do have a cacert setup in line 172 of this file, which is ,we get the cacert from the secret we created 
what's the difference between those code we added with previous logic?

Reply:
Openstack’s clouds.yaml has a cacert key referencing a path to a ca certificate file. 
For now CAPO only uses the ca certificate provided in the secret alongside the clouds.yaml file.
Here it adds support for reading file referenced in clouds.yaml.
It allows users to use containers’ ca certificates or ca certificates mounted from a configmap or a secret in CAPO pod.

Reply:
I provided an example in #1433